core/Common.php

```diff
@@ -256,6 +256,32 @@ public static function mb_strtoupper($string)
         return $string;
     }
 
+    /**
+     * Secure wrapper for unserialize, which by default disallows unserializing classes
+     *
+     * @param string $string String to unserialize
+     * @param array $allowedClasses Class names that should be allowed to unserialize
+     *
+     * @return mixed
+     */
+    public static function safe_unserialize($string, $allowedClasses = [])
+    {
+        if (PHP_MAJOR_VERSION >= 7) {
+            try {
+                return unserialize($string, ['allowed_classes' => empty($allowedClasses) ? false : $allowedClasses]);
+            } catch (\Throwable $e) {
+                $logger = StaticContainer::get('Psr\Log\LoggerInterface');
+                $logger->debug('Unable to unserialize a string: {message}', [
+                    'message' => $e->getMessage(),
+                    'backtrace' => $e->getTraceAsString()
+                ]);
+                return false;
```

Comment:
Can we log the string as well? For a debug log I presume it's ok.

Reply:
Added this myself, let me know if that's a problem (can remove in new PR if needed).